### ISSUE TITLE
Drop python 2.6 support

### ISSUE THREAD

**Issue by aabadie** [joblib/joblib#435]
I saw that [scikit-learn has recently dropped python 2.6 support](https://github.com/scikit-learn/scikit-learn/pull/7890) and joblib should do the same.

Several things to do (feel free to propose and amend the list):
- [ ] Remove the python 2.6 dependency in the CI matrix
- [ ] Drop the contextlib `closing()` dependency (mainly used in the persistence code)
- [ ] Remove the PY26 global variable
- [ ] Remove py26 pickle files in joblib/test/data

**Comment by GaelVaroquaux** [comment]
I was thinking the same actually.

**Comment by lesteve** [comment]
A tentative `git grep` to try to remove all the python 2.6 specific code:
```
❯ git grep -iP 'py(thon)?.*2.?6'
.travis.yml:    - PYTHON_VERSION="2.6" NUMPY_VERSION="1.6"
doc/installing.rst:* Installing only for a specific user is easy if you use Python 2.6 or
doc/memory.rst:  Beware that with Python 2.6 lambda functions cannot be separated out::
doc/persistence_fixture.py:"""Fixture module to skip the persistence doctest with python 2.6."""
doc/persistence_fixture.py:    if _compat.PY26:
doc/persistence_fixture.py:        # within a context manager (e.g in a `with` block) in python 2.6 so
doc/persistence_fixture.py:        raise SkipTest("Skipping persistence doctest in Python 2.6")
joblib/_compat.py:PY26 = sys.version_info[:2] == (2, 6)
joblib/func_inspect.py:    once we drop support for Python 2.6. We went for a simpler
joblib/func_inspect.py:    which is not available in Python 2.6.
joblib/numpy_pickle_utils.py:from ._compat import PY3_OR_LATER, PY26, PY27, _basestring
joblib/numpy_pickle_utils.py:    if PY26 or (PY27 and bz2 is not None and isinstance(fobj, bz2.BZ2File)):
joblib/numpy_pickle_utils.py:        # Python 2.6 doesn't fully support io.BufferedReader.
joblib/numpy_pickle_utils.py:    if PY26 or (PY27 and bz2 is not None and isinstance(fobj, bz2.BZ2File)):
joblib/numpy_pickle_utils.py:        # Python 2.6 doesn't fully support io.BufferedWriter.
joblib/numpy_pickle_utils.py:    versions of python >= 2.6.
joblib/numpy_pickle_utils.py:            if not PY26 and isinstance(data, memoryview):
joblib/numpy_pickle_utils.py:    Required as e.g. ZipExtFile in python 2.6 can return less data than
joblib/pool.py:    # customize the dispatch table without side effects in Python 2.6
joblib/test/data/generate_data.sh:CONDA_ENVS="py26-np16 py27-np17 py33-np18 py34-np19 py35-np19"
Binary file joblib/test/data/joblib_0.10.0_pickle_py26_np16.pkl.gzip matches
Binary file joblib/test/data/joblib_0.10.0_pickle_py35_np19.pkl.gzip matches
Binary file joblib/test/data/joblib_0.9.2_pickle_py26_np16.pkl matches
joblib/test/test_hashing.py:from joblib._compat import PY3_OR_LATER, PY26
joblib/test/test_hashing.py:if not PY26:
joblib/test/test_hashing.py:    expected_dict = {'py2': ['80f2387e7752abbda2658aafed49e086',
joblib/test/test_memory.py:        # catch_warnings is new in Python 2.6
joblib/test/test_memory.py:        # catch_warnings is new in Python 2.6
joblib/test/test_numpy_pickle.py:from joblib._compat import PY3_OR_LATER, PY26
joblib/test/test_numpy_pickle.py:    if not PY26:
joblib/testing.py:    # Python <= 2.6, we still need nose here
setup.py:              'Programming Language :: Python :: 2.6',

```
